Doc/howto/enum.rst

```diff
@@ -497,13 +497,23 @@ the :meth:`~Enum.__repr__` omits the inherited class' name.  For example::
     >>> Creature.DOG
     <Creature.DOG: size='medium', legs=4>
 
-Use the :func:`!dataclass` argument ``repr=False``
+Use the :func:`~dataclasses.dataclass` argument ``repr=False``
 to use the standard :func:`repr`.
 
 .. versionchanged:: 3.12
    Only the dataclass fields are shown in the value area, not the dataclass'
    name.
 
+.. note::
+
+   Adding :func:`~dataclasses.dataclass` decorator to :class:`Enum`
+   and its subclasses is not supported. It will not raise any errors,
+   but it will produce very strange results at runtime::
+
+      @dataclass  # don't do this: it does not make any sense
+      class Colors(Enum):
+         red = 'red'
```

Comment:
```suggestion
      >>> @dataclass                      # don't do this: it does not make any sense
      ... class Color(Enum):
      ...    RED = 1
      ...    GREEN = 2
      ...    BLUE = 3
      ...
      >>> Color.RED is Color.BLUE
      False
      >>> Color.RED == Color.BLUE
      True
```

Reply:
Do we *really* want to assert this behaviour that seems at least partially incorrect? 🤔

Reply:
The `==` is definitely incorrect, and showing it helps explain why using `@dataclass` on an enum is a bad idea.  We can certainly add a `# this should fail` to the `==` line, though.  Also, if that `==` test ever does work correctly, I want to know.  :smiley:

Reply:
Done 👍